braindecode/preprocessing/preprocess.py

```diff
@@ -117,11 +123,19 @@ def preprocess(concat_ds, preprocessors):
         A concat of BaseDataset or WindowsDataset datasets to be preprocessed.
     preprocessors: list(Preprocessor)
         List of Preprocessor objects to apply to the dataset.
+    save_dir : str | None
+        If a string, the preprocessed data will be saved under the specified
+        directory.
+    overwrite : bool
+        When `save_dir` is provided, controls whether to overwrite existing
+        files with the same name.
```

Comment:
This is not true. If `overwrite=True`, then all `fif` files (no matter whether epo or raw) in the directory are deleted as well as `target_name.json` and `description.json`. See https://github.com/braindecode/braindecode/blob/aecae2228af4f5d778c129154dfd7caeef3b11ab/braindecode/datasets/base.py#L320-L322.

Reply:
I have the impression we can really get into trouble with this saving and overwriting. Maybe we should discuss a bit more how to do this in a save way. One thing I can think of would be to enforce writing to an empty directory. This could be inconvenient for the user though...

Reply:
Right, we have to trust the user they will be careful... At least this should not overwrite raw data.

We could provide an additional "safer" option, so we would have the following possibilities:
1. Raise an error if there's already data under `save_dir`
2. Automatically create a new directory `save_dir + '_1'` (2, 3...) if the folder already exists
3. Overwrite